external/flume/src/test/scala/org/apache/spark/streaming/flume/FlumePollingStreamSuite.scala

```diff
@@ -167,26 +167,24 @@ class FlumePollingStreamSuite extends FunSuite with BeforeAndAfter with Logging
     }
   }
 
-  def writeAndVerify(channels: Seq[MemoryChannel], ssc: StreamingContext,
+  def writeAndVerify(sinks: Seq[SparkSink], channels: Seq[MemoryChannel], ssc: StreamingContext,
     outputBuffer: ArrayBuffer[Seq[SparkFlumeEvent]]) {
     val clock = ssc.scheduler.clock.asInstanceOf[ManualClock]
     val executor = Executors.newCachedThreadPool()
     val executorCompletion = new ExecutorCompletionService[Void](executor)
-    channels.map(channel => {
+
+    val latch = new CountDownLatch(batchCount * channels.size)
+    sinks.foreach(_.countdownWhenBatchReceived(latch))
+
+    channels.foreach(channel => {
       executorCompletion.submit(new TxnSubmitter(channel, clock))
     })
+
     for (i <- 0 until channels.size) {
       executorCompletion.take()
     }
-    val startTime = System.currentTimeMillis()
-    while (outputBuffer.size < batchCount * channels.size &&
-      System.currentTimeMillis() - startTime < 15000) {
-      logInfo("output.size = " + outputBuffer.size)
-      Thread.sleep(100)
-    }
-    val timeTaken = System.currentTimeMillis() - startTime
-    assert(timeTaken < 15000, "Operation timed out after " + timeTaken + " ms")
-    logInfo("Stopping context")
+
+    latch.await(15, TimeUnit.SECONDS)
```

Comment:
I see! I briefly looked at the code, I dont think we need to use manual clock or any of the complicated manual clock based TxnSubmitter. Instead why cant we do the FlumeStream approach, just ship some data, and wait with eventually to see all of them have been received or not.